### PR TITLE
#286: Removed Spray from various places

### PR DESCRIPTION
**Pull request by akara**
Thanks for your pull request.  Please review the following guidelines.

- [x] Title includes issue id.
- [x] Description of the change added.
- [x] Commits are squashed.
- [x] Tests added.
- [ ] Documentation added/updated.
- [x] Also please review [CONTRIBUTING.md](https://github.com/paypal/squbs/blob/master/CONTRIBUTING.md).

- Removed all Spray code from Unicomplex.
- Collapsed the tests from unicomplex/streaming into unicomplex.
- Re-build route and actor-based dispatching.
- Removed all Spray code from TestKit.
- Collapsed the tests from testkit/straming into testkit
- Removed spray.util usage from ActorMonitor and ActorRegistry.
- Removed squbs-admin from being a project in build.sbt, to be permanently replaced by the contents of squbs-admin-exp in the future.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -31,13 +31,13 @@ lazy val `squbs-pipeline` = project
...
 
-lazy val `squbs-unicomplex` = project dependsOn (`squbs-pipeline`, `squbs-streamingpipeline`)
+lazy val `squbs-unicomplex` = project dependsOn `squbs-streamingpipeline`
 
 lazy val `squbs-testkit` = project dependsOn `squbs-unicomplex`
 
 lazy val `squbs-zkcluster` = project dependsOn `squbs-testkit` % "test"
 
-lazy val `squbs-httpclient` = project dependsOn(`squbs-unicomplex`, `squbs-testkit` % "test")
+lazy val `squbs-httpclient` = project dependsOn(`squbs-unicomplex`, `squbs-pipeline`, `squbs-testkit` % "test")
```
Comment (akara) [inline build.sbt]: You're right. I did not prune the dependencies just yet. But I think the right approach is what you did with #308. Let me make this change as a separate PR. That might have even other dependencies to prune.

File: build.sbt [hunk 2]
```diff
@@ -67,8 +67,6 @@ lazy val `squbs-actorregistry` = project dependsOn (`squbs-unicomplex`, `squbs-t
 
 lazy val `squbs-actormonitor` = project dependsOn (`squbs-unicomplex`, `squbs-testkit` % "test")
 
-lazy val `squbs-admin` = project dependsOn (`squbs-unicomplex`, `squbs-testkit` % "test")
-
 lazy val `squbs-admin-exp` = project dependsOn (`squbs-unicomplex`, `squbs-testkit` % "test")
```
Comment (akara) [inline build.sbt]: Yes, enough churn with this one. That is just renaming a whole tree. Could be small. But the intention of this PR is just to remove Spray from Unicomplex. It has its ripple effects so I had to touch other projects. Let me file that PR, too.

File: squbs-testkit/build.sbt [hunk 3]
```diff
@@ -33,9 +31,9 @@ updateOptions := updateOptions.value.withCachedResolution(true)
 // SimpleTestKitTest is causing actor system conflict with SimpleTestKitSpec.  Forking logic is specific to
 // test name to minimize the duration of forked process (Travis CI limitations).
 def groupBySimpleTestKitTest(tests: Seq[TestDefinition]) =
-  tests groupBy (_.name == "org.squbs.testkit.japi.SimpleTestKitTest") map {
-    case (letter, tests) => new Tests.Group(letter.toString, tests, Tests.SubProcess(ForkOptions()))
-  } toSeq
+  tests .groupBy (_.name == "org.squbs.testkit.japi.SimpleTestKitTest") .map {
```
Comment (akara) [inline squbs-testkit/build.sbt]: OK. We'll have to tackle that.

File: build.sbt (hunk 2 above)
Comment (akara) [inline build.sbt]: #309

File: squbs-testkit/src/main/scala/org/squbs/testkit/TestRoute.scala [hunk 4]
```diff
@@ -1,65 +0,0 @@
-/*
```
Comment (akara) [inline squbs-testkit/src/main/scala/org/squbs/testkit/TestRoute.scala]: No, I just moved the tests. Should not be part of this PR, either. But there were a couple of tests that were wired to where they are and that made me take the step instead of trying to fix the test. That's the only reason src/test has been handled. Filed #310 for this.

File: squbs-testkit/src/test/scala/org/squbs/testkit/CustomTestKitSpec.scala [hunk 5]
```diff
@@ -17,31 +17,23 @@
...
-    val result = Await.result(pipeline(Get(s"http://127.0.0.1:${CustomTestKitSpec.port}/test")), awaitMax)
-    result.entity.asString should include ("success")
-  }
-
   it should "return OK with the port from PortGetter" in {
-    val pipeline = sendReceive
-    val result = Await.result(pipeline(Get(s"http://127.0.0.1:$port/test")), awaitMax)
-    result.entity.asString should include ("success")
+    implicit val materializer = ActorMaterializer()
+    val result = Await.result(entityAsString(s"http://127.0.0.1:$port/test"), awaitMax)
```
Comment (akara) [inline squbs-testkit/src/test/scala/org/squbs/testkit/CustomTestKitSpec.scala]: Yes, I won't go after another cleanup right now. But if you feel we should do it for 0.9, please do file a PR for that.

File: squbs-testkit/src/test/scala/org/squbs/testkit/CustomTestKitSpec.scala [hunk 6]
```diff
@@ -52,17 +44,13 @@ class CustomTestKitSpec extends CustomTestKit(CustomTestKitSpec.boot) with FlatS
...
-  val testConfig = ConfigFactory.parseMap(
-    Map(
-      "squbs.actorsystem-name" -> "CustomTestKitSpec",
-      "squbs.external-config-dir" -> "actorCalLogTestConfig",
-      "default-listener.bind-port" -> Int.box(port),
-      "squbs." + JMX.prefixConfig -> Boolean.box(true)
-    )
+  val testConfig = ConfigFactory.parseString(
+    s"""
+       |squbs.actorsystem-name = CustomTestKitSpec
```
Comment (akara) [inline squbs-testkit/src/test/scala/org/squbs/testkit/CustomTestKitSpec.scala]: I converted straight from the map. Let me know if you feel strongly. I can fix it. No big deal.

File: squbs-testkit/src/test/scala/org/squbs/testkit/package.scala [hunk 7]
```diff
@@ -16,7 +16,30 @@
...
+import akka.stream.ActorMaterializer
+import akka.util.ByteString
+
+import scala.concurrent.Future
+
 package object testkit {
   case object TestPing
   case object TestPong
+
+  def entityAsString(uri: String)(implicit am: ActorMaterializer, system: ActorSystem): Future[String] = {
```
Comment (akara) [inline squbs-testkit/src/test/scala/org/squbs/testkit/package.scala]: That's OK. There's more to cleanup. We could convert the entity toStrict instead. I'm still trying to find a good strategy for this.

File: squbs-unicomplex/src/main/scala/org/squbs/unicomplex/ServiceRegistryBase.scala [hunk 8]
```diff
@@ -31,13 +31,13 @@ trait ServiceRegistryBase[A] {
 
   val log: LoggingAdapter
 
-  protected def listenerRoutes: Map[String, Seq[(A, ServiceHandlerWrapper, PipelineSetting)]]
+  protected def listenerRoutes: Map[String, Seq[(A, FlowWrapper, PipelineSetting)]]
```
Comment (akara) [inline squbs-unicomplex/src/main/scala/org/squbs/unicomplex/ServiceRegistryBase.scala]: Yes, I got rid of `ServiceHandlerWrapper` and `ActorWrapper` altogether. Everything is represented by `FlowWrapper` which has the `Flow` and a responsible `ActorRef` for that flow.

File: squbs-unicomplex/src/main/scala/org/squbs/unicomplex/Unicomplex.scala [hunk 9]
```diff
@@ -662,36 +653,18 @@ class CubeSupervisor extends Actor with ActorLogging with GracefulStopHelper {
       }
 
     case StartCubeService(webContext, listeners, props, name, proxyName, ps, initRequired) =>
+      val hostActor = context.actorOf(props, name)
+      import ConfigUtil._
+      val concurrency = context.system.settings.config.get[Int]("akka.http.server.pipelining-limit")
```
Comment (akara) [inline squbs-unicomplex/src/main/scala/org/squbs/unicomplex/Unicomplex.scala]: Perhaps the right name is `parallelism`? That would be the term used by the `mapAsync` API.

File: squbs-unicomplex/src/main/scala/org/squbs/unicomplex/Unicomplex.scala [hunk 10]
```diff
@@ -662,36 +653,18 @@ class CubeSupervisor extends Actor with ActorLogging with GracefulStopHelper {
...
+      val flow = Flow[HttpRequest].mapAsync(concurrency) { request => (hostActor ? request).mapTo[HttpResponse] }
```
Comment (akara) [inline squbs-unicomplex/src/main/scala/org/squbs/unicomplex/Unicomplex.scala]: Yes, right on. This case match will just disappear in 1.0 and we have to remove the `isAssignableFrom` check from the case match above it.

File: squbs-unicomplex/src/test/scala/org/squbs/unicomplex/CubeActorErrorStatesSpec.scala [hunk 11]
```diff
@@ -1,52 +1,45 @@
 /*
- *  Copyright 2015 PayPal
```
Comment (akara) [inline squbs-unicomplex/src/test/scala/org/squbs/unicomplex/CubeActorErrorStatesSpec.scala]: Eh? This is also formatted by my IDE. I have not touched these, at least not intentionally.

File: squbs-unicomplex/src/test/scala/org/squbs/unicomplex/ListenerStateSpec.scala [hunk 12]
```diff
@@ -20,21 +20,20 @@ import javax.management.ObjectName
...
-import spray.util.Utils
+import org.squbs.unicomplex.streaming.RouteDefinition
 
 object ListenerStateSpec{
 
   val classPaths = Array(getClass.getClassLoader.getResource("classpaths/ListenerState").getPath)
 
-  val (_, port) = Utils.temporaryServerHostnameAndPort()
-  val (_, port2) = Utils.temporaryServerHostnameAndPort()
+  val (_, _, port) = temporaryServerHostnameAndPort()
```
Comment (akara) [inline squbs-unicomplex/src/test/scala/org/squbs/unicomplex/ListenerStateSpec.scala]: Yes. One of the issues I filed should address this. Don't remember which one. I did not exactly reference this test but it is on similar grounds.

File: squbs-unicomplex/src/test/resources/classpaths/DummyFailedFlowSvc1/META-INF/squbs-meta.conf [hunk 13]
```diff
@@ -0,0 +1,8 @@
+cube-name = org.squbs.unicomplex.test.DummyFailedFlowSvc1
```
Comment (akara) [inline squbs-unicomplex/src/test/resources/classpaths/DummyFailedFlowSvc1/META-INF/squbs-meta.conf]: It is the same file, just all in Streaming. This is added recently when I added the `FlowDefinition`.

File: squbs-unicomplex/src/main/scala/org/squbs/unicomplex/Unicomplex.scala (hunk 9 above)
Comment (anilgursel) [inline squbs-unicomplex/src/main/scala/org/squbs/unicomplex/Unicomplex.scala]: I agree
